app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,101 @@
+class RepositoriesController < ApplicationController
+  def index
+    @repositories = Repository.where(hide: false).paginate(:page => params[:page])
+    respond_to do |format|
+        format.html
+        format.js
+    end
+  end
+
+  def show
+    @repository = Repository.find(params[:id])
+    impressionist(@repository, nil, { unique: [:session_hash] })
+  end
+
+  def new
```

Comment:
@AmitPatel-BoTreeConsulting  sir, This method is use for creating new Repository and while initialize the new object of Repository set some default values that comes from github.